src/api/store/userprofile.py

```diff
@@ -432,6 +433,9 @@ def create_user(self, context: Context, args) -> Tuple[dict, MassEnergizeAPIErro
 
       new_user_email = False
       existing_user = UserProfile.objects.filter(email=email).first()
+      if not existing_user and not is_new_guest_allowed:
```

Comment:
ditto.  the feature flag should control this without the back-end needing this logic.